src/pcre2_compile_class.c

```diff
@@ -147,8 +148,23 @@ while (c <= end)
   /* Compute caseless set. */
   co = UCD_CASESET(c);
 
-  if (co != 0 && (!(options & PARSE_CLASS_RESTRICTED_UTF)
-                  || PRIV(ucd_caseless_sets)[co] > 127))
+  if ((options & PARSE_CLASS_TURKISH_UTF) && UCD_ANY_I(c))
+    {
+    list = tmp;
```

Comment:
You could simply set the `co` variable. Less code duplication.

Reply:
Ah yes! That goes back to an earlier version of the code where I didn't have a caseset for the Turkish ones.